src/main/java/org/openrewrite/java/testing/junit5/ExpectedExceptionToAssertThrows.java

```diff
@@ -93,59 +116,128 @@ public String getDescription() {
+                    isExpectMessageArgAMatcher = isHamcrestMatcher((J.MethodInvocation) expectMessageMethodArg);
+                    if (!isExpectMessageArgAMatcher) {
+                        return m;
+                    }
+                } else {
+                    if (!(expectMessageMethodArg instanceof J.Literal &&
+                            expectMessageMethodArg.getType() == JavaType.Primitive.String)) {
+                        return m;
+                    }
+                }
+            }
 
-                StringBuilder printedStatements = new StringBuilder();
-                for (Statement stmt : statements) {
-                    printedStatements.append(stmt.print()).append(';');
+            boolean isExpectedCauseArgAMatcher = false;
+            if (expectCauseMethodInvocation != null) {
+                List<Expression> args = expectCauseMethodInvocation.getArguments();
+                if (args.size() != 1) {
+                    return m;
                 }
-                J.MethodInvocation expectedMessageMethodInvocation = getCursor().pollMessage(EXPECT_MESSAGE_INVOCATION_KEY);
-                if (expectedMessageMethodInvocation == null) {
-                    m = m.withBody(
-                            m.getBody().withTemplate(
-                                    template("{ assertThrows(#{}, () -> { #{} }); }")
-                                            .javaParser(ASSERTIONS_PARSER.get())
-                                            .staticImports("org.junit.jupiter.api.Assertions.assertThrows")
-                                            .build(),
-                                    m.getBody().getCoordinates().replace(),
-                                    expectedException, printedStatements.toString())
-                    );
-                } else {
-                    String expectedMessage = expectedMessageMethodInvocation.getArguments().get(0).print();
-                    m = m.withBody(
-                            m.getBody().withTemplate(
-                                    template("{ assertThrows(#{}, () -> { #{} }, #{}); }")
-                                            .javaParser(ASSERTIONS_PARSER.get())
-                                            .staticImports("org.junit.jupiter.api.Assertions.assertThrows")
-                                            .build(),
-                                    m.getBody().getCoordinates().replace(),
-                                    expectedException, printedStatements.toString(), expectedMessage));
+
+                final Expression expectCauseMethodArg = args.get(0);
+                if (expectCauseMethodArg instanceof J.MethodInvocation) {
+                    isExpectedCauseArgAMatcher = isHamcrestMatcher((J.MethodInvocation) expectCauseMethodArg);
+                    if (!isExpectedCauseArgAMatcher) {
+                        return m;
+                    }
```

Comment:
Same story here, if things don't match up, we don't execute the template, maybe this is ok and by design? just wanted to point it out.